aggregator_core/src/batch_mode.rs

```diff
@@ -370,9 +370,12 @@ impl Iterator for TimeIntervalBatchIdentifierIter {
         // batch interval used to create the iterator.
         let interval = Interval::new(
             self.start
-                .add(&Duration::from_seconds(
-                    self.step * self.time_precision.as_seconds(),
-                ))
+                .add_timedelta(
+                    &chrono::TimeDelta::try_seconds(
+                        (self.step * self.time_precision.as_seconds()) as i64,
```

Comment:
Some of these casts make me a bit nervous. I wonder if we could add extension method like `try_seconds_unsigned` or just a `TryFrom<u64>` impl on `TimeDelta` that handle them safely, improving the ergonomics at all the callsites.